plugins/teststeps/teststeps.go

```diff
@@ -29,94 +30,66 @@ type PerTargetFunc func(cancel, pause <-chan struct{}, target *target.Target) er
 // each target. The implementation of the per-target function is responsible for
 // handling internal cancellation and pausing.
 func ForEachTarget(pluginName string, cancel, pause <-chan struct{}, ch test.TestStepChannels, f PerTargetFunc) error {
-	type tgtErr struct {
-		target *target.Target
-		err    error
+	// Temporary solution
+	ctx, cancelCtx := combineChannelsToContext(cancel, pause)
```

Comment:
All code that makes  context creation/cancelling will disappear during PR that introduces statectx.Context instead of pause/cancel channels in teststeps plugins.